### ISSUE TITLE
No warnings about setting a deprecated property

### ISSUE THREAD

**Issue by maranomynet** [gund/eslint-plugin-deprecation#13]
```ts
type T = {
	new?: string;
	/** @deprecated */
	old?: string;
};
const myT: T = { 
	// I'd expcect an ESLint warning here, but there is none
	old: 'value'
};

console.info(myT.old); // I only get a warning here
```

How it looks in VSCode:
<img width="520" alt="image" src="https://user-images.githubusercontent.com/86827/95744583-135c5780-0c83-11eb-9817-68b8b8f7cf69.png">

In fact VSCode doesn't warn (with a strike-through) about this either.

**Comment by remLse** [comment]
This may actually be related to an issue with TypeScript itself. It would possibly be enough to wait for a bug fix and then make sure the latest version of TypeScript is used with this plugin. Fingers crossed.

https://github.com/microsoft/TypeScript/issues/39374

**Comment by gund** [comment]
We just released an updated with latest deps in v1.3.0 maybe you can try it out and see if this is still an issue?

**Comment by stekern** [comment]
I seem to be having the same issue in v1.3.2 using TypeScript v4.5.4.

The plugin correctly picks up deprecated methods, but nothing seems to be reported for deprecated object properties.

**Comment by gund** [comment]
Actually now that I look at this - it makes total sense as the plugin is only reporting usages of deprecated code.

And in your case declaring an object with deprecated property is not usage it's declaration so it's not reported.

**Comment by stekern** [comment]
Ah, okay. I was under the impression that the plugin would pick up everything annotated with `@deprecated`, be it methods or properties.

I do still think it would be beneficial to have this functionality for quickly and easily being made aware of usage of deprecated properties in our codebases. Is this functionality you'd consider in scope for this plugin?

**Comment by maranomynet** [comment]
Yeah, that distinction does not seem useful to me.

When I declare an object of a certain type, I'm **using** it's interface-/type-delcaration.

For example: When I pass a deprecated prop on a React component, I'm in-place declaring the props object and get no warning that I'm using (in way of passing) a prop that the component's author explicitly flagged as `@deprecated`.

The way things are now, the only person to ever see a warning of a deprecated prop/option is the library author who flagged the prop themself, but is forced to use those props to maintain back-compatibility.

**Comment by gund** [comment]
Hmm this might actually be a valid use-case.
I will reopen the issue to track it and PRs are welcome 😊

**Comment by nelson6e65** [comment]
This plugin does not detect deprecated properties when the definition is in another file.

**Comment by mpinkard-jsq** [comment]
I am wondering if there is any progress on this issue? We are trying to use this plugin in a large React project to deprecate props for certain components. Depending on the complexity, we may be interested in submitting a PR, but we would need some guidance

**Comment by gund** [comment]
I don't think there was any meaningful progress to fix this bug, so you are more then welcome to try and figure a fix for it in a PR.
Unfortunately I could not be of help as I was not digging into eslint plugins (yet), maybe somebody else here can help but tbh I have no clue.
One thing I can suggest as a helping tool is to use some [TS AST explorer](https://ts-ast-viewer.com/#) which could give you an idea how the fix could be implemented.

**Comment by sebakerckhof** [comment]
Would this be possible by now? Since it seems that TS knows about react props that are deprecated, but I get no warnings for those from eslint:
![image](https://github.com/gund/eslint-plugin-deprecation/assets/88471/c0be90de-c625-46c4-a380-49afaf7172d3)

**Comment by sebakerckhof** [comment]
Would this be possible by now? Since it seems TS knows this is deprecated, but eslint doesn't warn me:
![image](https://github.com/gund/eslint-plugin-deprecation/assets/88471/9e53a1d2-858c-4b0e-ada7-d5364b3729bc)
